cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/remote/RemoteExecutionTwillPreparer.java

```diff
@@ -0,0 +1,877 @@
+  private final List<String> applicationClassPaths = new ArrayList<>();
+  //private final Credentials credentials;
+  private final Map<String, Map<String, String>> logLevels = new HashMap<>();
+  private final LocationCache locationCache;
+  private final Map<String, Integer> maxRetries = new HashMap<>();
+  private final Map<String, Map<String, String>> runnableConfigs = new HashMap<>();
+  private final Map<String, String> runnableExtraOptions = new HashMap<>();
+  private final SSHConfig sshConfig;
+  private final ProgramRunId programRunId;
+  private final LocationFactory locationFactory;
+  private final MessagingService messagingService;
+  private String extraOptions;
+  private JvmOptions.DebugOptions debugOptions;
+
+  private ClassAcceptor classAcceptor;
+  private String classLoaderClassName;
+
+  RemoteExecutionTwillPreparer(CConfiguration cConf, Configuration hConf, SSHConfig sshConfig,
+                               ProgramRunId programRunId, TwillSpecification twillSpec,
+                               RunId runId, @Nullable String extraOptions,
+                               LocationCache locationCache, LocationFactory locationFactory,
+                               MessagingService messagingService) {
+    // Check to prevent future mistake
+    if (twillSpec.getRunnables().size() != 1) {
+      throw new IllegalArgumentException("Only one TwillRunnable is supported");
+    }
+
+    this.debugOptions = JvmOptions.DebugOptions.NO_DEBUG;
+    this.cConf = cConf;
+    this.hConf = hConf;
+    this.sshConfig = sshConfig;
+    this.programRunId = programRunId;
+    this.twillSpec = twillSpec;
+    this.runId = runId;
+    this.extraOptions = extraOptions == null ? "" : extraOptions;
+    this.classAcceptor = new ClassAcceptor();
+    this.locationCache = locationCache;
+    this.locationFactory = locationFactory;
+    this.messagingService = messagingService;
+  }
+
+  private void confirmRunnableName(String runnableName) {
+    Preconditions.checkNotNull(runnableName);
+    Preconditions.checkArgument(twillSpec.getRunnables().containsKey(runnableName),
+                                "Runnable %s is not defined in the application.", runnableName);
+  }
+
+  @Override
+  public TwillPreparer withConfiguration(Map<String, String> config) {
+    for (Map.Entry<String, String> entry : config.entrySet()) {
+      this.hConf.set(entry.getKey(), entry.getValue());
+    }
+    return this;
+  }
+
+  @Override
+  public TwillPreparer withConfiguration(String runnableName, Map<String, String> config) {
+    confirmRunnableName(runnableName);
+    runnableConfigs.put(runnableName, Maps.newHashMap(config));
+    return this;
+  }
+
+  @Override
+  public TwillPreparer addLogHandler(LogHandler handler) {
+    LOG.trace("LogHandler is not supported for {}", getClass().getSimpleName());
+    return this;
+  }
+
+  @Override
+  public TwillPreparer setUser(String user) {
+    return this;
```

Comment:
do we want a trace log for this too?

Reply:
This method is actually already deprecated in Twill and we are not using it. Need to implement it for the sake of implementing an interface.